code/modules/mob/living/simple_animal/vore/sarlaac.dm

```diff
@@ -0,0 +1,239 @@
+//////////////////////////////////////////////////////////////////////////////
+
+/*
+/mob/living/simple_animal/hostile/testmob/ClosestDistance() // Needed or the SA AI won't ever try to teleport.
+	if(world.time > lastspecialattack + special_rangedcooldown)
+		return spattack_max_range - 1
+	return ..()
+*/
+
+/mob/living/simple_animal/hostile/testmob/SpecialAtkTarget()
+	// Teleport attack.
+	if(!target_mob)
+		to_chat(src, "<span class='warning'>There's nothing to teleport to.</span>")
+		return FALSE
+	var/list/nearby_things = range(1, target_mob)
+	var/list/valid_turfs = list()
+
+	// All this work to just go to a non-dense tile.
+	for(var/turf/potential_turf in nearby_things)
+		var/valid_turf = TRUE
+		if(potential_turf.density)
+			continue
+		for(var/atom/movable/AM in potential_turf)
+			if(AM.density)
+				valid_turf = FALSE
+		if(valid_turf)
+			valid_turfs.Add(potential_turf)
+
+	//if(special_rangedcooldown <= world.time + ranged_cooldown_time*0.25 && !pre_attack)
+	if(world.time < lastspecialattack + special_rangedcooldown)
+		to_chat(src, "<span class='warning'>You can't teleport right now, wait a few seconds.</span>")
+		return FALSE
+	var/turf/target_turf = pick(valid_turfs)
+	if(!target_turf)
+		to_chat(src, "<span class='warning'>no</span>")
+	playsound(src, 'sound/weapons/heavysmash.ogg', 50, 1)
+	visible_message("<span class='danger'>[target_mob] has been throw by [src] tentacles closer to itself!</span>")
+	target_mob.Weaken(10)
+	if(prob(30))
+		target_mob.Stun(100)
+	target_mob.adjustBruteLoss(rand(10,15))
+	target_mob.throw_at(pick(target_turf), 10, 3)	//All this to throw them near mouth.
+	sleep(20)	//Grab ur friend
+	lastspecialattack = world.time
+
+/*
+/mob/living/simple_animal/hostile/testmob/proc/handle_preattack()
+	if(ranged_cooldown <= world.time + ranged_cooldown_time*0.25 && !pre_attack)
+		pre_attack++
+	if(!pre_attack || stat || AIStatus == AI_IDLE)
+		return
+	icon_state = pre_attack_icon
+
+/mob/living/simple_animal/hostile/testmob/proc/handle_tentacles()
+	if(stance == STANCE_IDLE)
+		var/tturf = get_turf(target)
+		if(!isturf(tturf))
+			return
+		if(get_dist(src, target) <= 3)//Screen range check, so you can't get tentacle'd offscreen
+			visible_message("<span class='warning'>[src] digs its tentacles under [target]!</span>")
+			new /obj/effect/temp_visual/goliath_tentacle/original(tturf, src)
+			ranged_cooldown = world.time + ranged_cooldown_time
+			icon_state = icon_aggro
+			pre_attack = 0*/
+
+
+//////////////////////////////////////////////////////////////////////////////
+//////////////////////////////////////////////////////////////////////////////
+
+//////////////////////////////////////////////////////////////////////////////
+//////////////////////////////////////////////////////////////////////////////
+
+
+// Make sure you don't call ..() on this one, otherwise you duplicate work.
+/mob/living/simple_animal/init_vore()
```

Comment:
But... this gives EVERY s_a two bellies. Don't do that. :v

Reply:
Confirmed with a catgirl. I'll fix.